uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/PathUtil.java

```diff
@@ -66,8 +66,7 @@ public String stripRepoNameAndSpace(String strippedPath) {
      */
     public String getNiogitRepoPath(org.uberfire.java.nio.file.Path path) {
         try {
-            final JGitPathImpl gPath = (JGitPathImpl) path;
-            final File directory = gPath.getFileSystem().getGit().getRepository().getDirectory();
+            final File directory = path.toFile();
```

Comment:
@rhtevan Yes it will because that same behavior is hidden behind that JGitPath implementation instead of being explicitly casted. So if a UnixPath or a JGitPath path is called, they will work the same with its own implementation.

Reply:
Due to [this change](https://github.com/kiegroup/appformer/blob/aede85844c2481fb1f2f5a4752bf8d5e93e42cf6/uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemImpl.java#L254-L255), the path implementation will be an instance of 'JGitFSPath', and its '.[toFile](https://github.com/adrielparedes/appformer/blob/aede85844c2481fb1f2f5a4752bf8d5e93e42cf6/uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFSPath.java#L151)' implementation has been changed to 
```
return fs.getGit().getRepository().getDirectory();
```
It's the same with original implementation.